docs/utils.py

```diff
@@ -0,0 +1,235 @@
+    markdown_split_str.append("")  # add last empty line
+    markdown_str = "\n".join(markdown_split_str)
+
+    with open(md_output_filepath, "w", encoding="utf-8") as file:
+        file.write(markdown_str)
+
+
+def is_markdown_up_to_date(
+    ipynb_filepath: Path, md_filepath: Path, remove_cell_tags: Sequence
+) -> bool:
+    """Check if markdown file is up to date with its associated notebook."""
+    assert ipynb_filepath.is_file(), f"{ipynb_filepath} does not exist."
+    assert md_filepath.is_file(), f"{md_filepath} does not exist."
+
+    with NamedTemporaryFile(suffix=".md") as temp_file:
+        notebook_to_markdown(
+            ipynb_filepath=ipynb_filepath,
+            md_output_filepath=Path(temp_file.name),
+            remove_cell_tags=remove_cell_tags,
+        )
+        with open(temp_file.name, "r", encoding="utf-8") as file:
+            temp_markdown = file.read()
+    with open(md_filepath, "r", encoding="utf-8") as file:
+        markdown = file.read()
+
+    return temp_markdown == markdown
+
+
+class OutdatedMarkdownError(Exception):
+    """Raised when markdown is not up to date with notebook."""
+
+
+class OutdatedMkDocs(Exception):
+    """Raised when mkdocs.yml is not up to date."""
+
+
+@click.command(name="notebook_tutorials_commit_hook")
+@click.argument(
+    "modified_files",
+    type=click.Path(exists=True, dir_okay=False, path_type=Path),
+    required=True,
+    nargs=-1,
+)
+def notebook_tutorials_commit_hook(modified_files: Sequence[Path]):
+    """
+    Check if all notebooks in staging are up-to-date with their markdown files.
+
+    Assumes that markdown files are in the same directory as notebooks, and have the same name.
+
+    Also checks if markdown files are in mkdocs.yml.
+    """
+    groupby_iter = groupby(modified_files, key=lambda path: path.stem)
+    for filename, group in groupby_iter:
+        group = list(group)
+        # check only notebooks modified in docs/sdk/tutorials
+        if "docs/sdk/tutorials" not in str(group[0].parent):
+            continue
+        # skip single markdown files, probably hand written tutorials without notebooks
+        if len(group) == 1 and group[0].suffix == ".md":
+            continue
+        if len(group) != 2:
+            raise ValueError(
+                f"Expected two files (.md and .ipynb) in staging for '{filename}', got {group}."
+            )
+        ipynb_filepath, md_filepath = sorted(group)
+        assert ipynb_filepath.suffix == ".ipynb", ipynb_filepath
+        assert md_filepath.suffix == ".md", md_filepath
+
+        ipynb_filepath = ipynb_filepath.resolve()
+        md_filepath = md_filepath.resolve()
+
+        if not is_markdown_up_to_date(
+            ipynb_filepath=ipynb_filepath,
+            md_filepath=md_filepath,
+            remove_cell_tags=DEFAULT_REMOVE_CELL_TAGS,
+        ):
+            raise OutdatedMarkdownError(f"{ipynb_filepath} is not up to date with {md_filepath}.")
```

Comment:
why do you raise an error instead of replacing the markdown with the one that you generated to test the equality?

Reply:
like this, we wouldn't need to call notebook_to_markdown_cmd manually.

Reply:
I understand. But then we could not preview the markdown on Mkdocs to assert that everything is good. If we are currently not sure of the quality of generation, I am ok to begin to generate markdown manually with the command and then do it manually if we are more confident